src/Rules/Comparison/ImpossibleCheckTypeHelper.php

```diff
@@ -78,6 +80,19 @@ public function findSpecifiedType(
 				], true)) {
 					return null;
 				}
+				if ($functionName === 'array_is_list' && count($node->getArgs()) === 1) {
+					$arrayValue = $scope->getType($node->getArgs()[0]->value);
+
+					if (!$arrayValue instanceof ArrayType) {
+						return null;
+					}
+
+					if ((new StringType())->isSuperTypeOf($arrayValue->getKeyType())->yes()) {
+						return false;
+					}
+
+					return null;
+				}
```

Comment:
I am wondering whether these changes are required?
isn't the type-specifying extension enough to get it working?